interface/forms/questionnaire_assessments/questionnaire_assessments.php

```diff
@@ -0,0 +1,191 @@
+                        if (typeof data.copyrightNotice !== 'undefined' && data.copyrightNotice > '') {
+                            document.getElementById('copyright').value = jsAttr(data.copyrightNotice);
+                            document.getElementById('copyrightNotice').innerHTML = jsText(data.copyrightNotice);
+                        }
+                        LForms.Util.addFormToPage(data, 'formContainer');
+                        return data;
+                    }).then((data) => {
+                        let qFhir = LForms.Util.getFormFHIRData("Questionnaire", 'R4', data);
+                        document.getElementById('questionnaire').value = JSON.stringify(qFhir);
+                    });
+                }
+            });
+        }
+    </script>
+</head>
+<body>
+    <div class="container-xl my-2">
+        <form method="post" id="qa_form" name="qa_form" onsubmit="return saveQR(this)" action="<?php echo $rootdir; ?>/forms/questionnaire_assessments/save.php?form_id=<?php echo attr_url($formid) ?>">
+            <input type="hidden" name="csrf_token_form" value="<?php echo attr(CsrfUtils::collectCsrfToken()); ?>" />
+            <input type="hidden" id="lform" name="lform" value="<?php echo attr($form['lform'] ?: ''); ?>" />
+            <input type="hidden" id="lform_response" name="lform_response" value="<?php echo attr($form['lform_response'] ?: ''); ?>" />
+            <input type="hidden" id="code_type" name="code_type" value="<?php echo attr($form['code_type'] ?: ''); ?>" />
+            <input type="hidden" id="code" name="code" value="<?php echo attr($form['code'] ?: ''); ?>" />
+            <input type="hidden" id="copyright" name="copyright" value="<?php echo attr($form['copyright'] ?: ''); ?>" />
+            <input type="hidden" id="questionnaire" name="questionnaire" value="<?php echo attr($form['questionnaire'] ?: ''); ?>" />
+            <input type="hidden" id="questionnaire_response" name="questionnaire_response" value="<?php echo attr($form['questionnaire_response'] ?: ''); ?>" />
+            <div class="form-group">
+                <div class="input-group isNew">
+                    <label for="loinc_item" class="font-weight-bold mt-2 mr-1"><?php echo xlt("Search and Select a LOINC form") . ': '; ?></label>
+                    <input class="form-control search_field" type="text" id="loinc_item" placeholder="<?php echo xla("Type to search"); ?>" autocomplete="off" role="combobox" aria-expanded="false">
+                </div>
+                <div>
+                    <p class="text-center"><?php echo "<span class='font-weight-bold'>" . xlt("Note") . ": </span><i>" . xlt("LOINC form definitions are subject to the LOINC"); ?>
+                        <a href="http://loinc.org/terms-of-use" target="_blank"><?php echo xlt("terms of use.") . "</i>"; ?></php></a></p>
+                    <p id="copyrightNotice">
+                        <?php echo text($form['copyright'] ?: ''); ?>
```

Comment:
would only use elvis operator when it is needed since it will still throw a php warning when empty (unlike use of `::` which will not throw a php warning when empty)

Reply:
All coalesce does is prevent undefined (strongly typed in c) from throwing. A purposeful setting to null should not throw. I don't want null sent to interface thus the conditional! copyright in DB is default null.

Reply:
however you're correct here because a new questionnaire is undefined form rows. sorry, I appear to not be with it lately!
I did notice that we are in for some trouble soon with deprecated casting in areas e.g
`[09-Aug-2022 10:08:19 America/New_York] PHP Deprecated:  Automatic conversion of false to array is deprecated in C:\xampp\htdocs\openemr\src\Gacl\Gacl.php on line 569`

soon we may need to get all of together to turn on deprecated warnings and errors and mass fix these.

I/we have a theme issue with lforms as it is themed to BS3. I'd have to fork and refactor lform project and contribute back to project or we maintain our version of lforms. I know how to do if my brain would cooperate but unsure if I want to tackle just now!